libs/constrained-generators/src/Constrained/Spec/SumProd.hs

```diff
@@ -472,15 +498,35 @@ branchW ::
 branchW w body =
   Weighted (Just w) (bind (buildBranch @p body . toArgs @a))
 
+type ProdAsListComputes a =
+  AssertSpineComputes
+    (Text "Have you considered adding an IsNormalType or ProdAsListComputes constraint?")
+    (ProductAsList a)
+
+-- ProdAsListComputes is here to make sure situations like this:
```

Comment:
Comment seems unfinished: "... situations like this", because it is hard to see that "Get reasonable type errors prompting you like this:" is the actual continuation.

I'd suggest moving the whole comment into haddock for `ProdAsListComputes` with addition of `@@@` syntax that wraps code snippets, which would make this distinction between code snippet and actual description more clear.

Reply:
Sorted!